spec/miq_ae_playbook_method_spec.rb

```diff
@@ -0,0 +1,152 @@
+describe MiqAeEngine::MiqAePlaybookMethod do
+  describe "run" do
+    let(:user) { FactoryGirl.create(:user_with_group) }
+    let(:aw) { FactoryGirl.create(:automate_workspace, :user => user, :tenant => user.current_tenant) }
+    let(:root_hash) { { 'name' => 'Flintstone' } }
+    let(:root_object) { Spec::Support::MiqAeMockObject.new(root_hash) }
+    let(:persist_hash) { {} }
+    let(:script) { {"a" => "1"} }
+    let(:method_name) { "Freddy Kreuger" }
+    let(:method_key) { "FreddyKreuger_ansible_method_task_id" }
+    let(:miq_task) { FactoryGirl.create(:miq_task) }
+
+    let(:workspace) do
+      double("MiqAeEngine::MiqAeWorkspaceRuntime", :root               => root_object,
+                                                   :persist_state_hash => persist_hash,
+                                                   :ae_user            => user,
+                                                   :hash_workspace     => {},
+                                                   :current_namespace  => "Bedrock",
+                                                   :current_class      => "Mogul",
+                                                   :current_instance   => "Fred",
+                                                   :current_message    => "Yabba Dabba",
+                                                   :current_method     => "Dino")
+    end
+
+    let(:user) do
+      FactoryGirl.create(:user_with_group, :userid   => "admin",
+                                           :settings => {:display => { :timezone => "UTC"}})
+    end
+
+    let(:aem)    { double("AEM", :data => script.to_yaml, :name => method_name) }
+    let(:obj)    { double("OBJ", :workspace => workspace) }
+    let(:inputs) { { 'name' => 'Fred' } }
+
+    context "regular method" do
+      it "success" do
+        allow(described_class::PLAYBOOK_CLASS).to receive(:run).and_return(miq_task.id)
+        allow(MiqRegion).to receive(:my_region).and_return(FactoryGirl.create(:miq_region))
+        allow(AutomateWorkspace).to receive(:create).and_return(aw)
+        miq_task.update_status(MiqTask::STATE_FINISHED, MiqTask::STATUS_OK, "Done")
+        allow(MiqTask).to receive(:wait_for_taskid).and_return(miq_task)
+        allow(workspace).to receive(:update_workspace)
```

Comment:
@mkanoor This basic pattern repeat 6 times in these tests.  Can you DRY up this logic?